### PR TITLE
Merge Kohl's Custom Changes Into Master Branch

### PR DESCRIPTION
**Pull request by seanmalloy**

### CODE REVIEW

File: tests/test-openshift_provision-generate_resource.yml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+  - name: Verify Service
+    fail:
+      msg: |
+        Service not defined as expected
+        >>>
+        {{ cmp_service | to_yaml }}
+        ===
+        {{ got_service | to_yaml }}
+        <<<
```
Comment (seanmalloy) [inline tests/test-openshift_provision-generate_resource.yml:60]: Line 56, 58, and 60 look messed up to me.

File: library/openshift_provision.py [hunk 2]
```diff
@@ -230,9 +230,12 @@ def compare_set_list(path, src, dst):
...
 def set_dict_defaults(d, default):
-    for k, v in default.items():
-        if k not in d:
-            d[k] = v
+    if d is None:
+        d = default.items()
+    else: 
+        for k, v in default.items():
+            if k not in d:
+                d[k] = v
```
Comment (seanmalloy) [inline library/openshift_provision.py:238]: This block of code should fix https://github.com/gnuthought/ansible-role-openshift-provision/issues/53

File: tests/test-openshift_provision-generate_resource.yml (hunk 1 above)
Comment (vinny-sabatini) [reply tests/test-openshift_provision-generate_resource.yml:60]: @seanmalloy I don't believe this is messed up, I believe I had used a different test as an example.
Ref: https://github.com/KohlsTechnology/ansible-role-openshift-provision/blob/6db8c0e505138ebdd4a7d2d5c91861f745d519e2/tests/test-cluster_resources-ClusterResourceQuota.yml#L94-L102

**Review by vinny-sabatini — APPROVED** [review]

File: tests/test-openshift_provision-generate_resource.yml (hunk 1 above)
Comment (seanmalloy) [reply tests/test-openshift_provision-generate_resource.yml:60]: @vinny-sabatini sounds good

**Comment by vinny-sabatini** [comment]
It looks like the checks are failing because there is not a `tests/test.yml` file. 
That file was removed in November 2017 - https://github.com/KohlsTechnology/ansible-role-openshift-provision/commit/4c4cafab42e8878dad2e7ce13010180c48263a4a

**Pull request merged**